src/main/java/com/google/opensesame/servlets/UserServlet.java

```diff
@@ -31,19 +35,51 @@ public void init() {
   }
 
   @Override
-  // Get a specific user. Return null if not found. TODO: User Validation
+  // Get a specific user. Return the currently signed in user if no userID is
+  // supplied.
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String userGithub = request.getParameter("githubID");
-    Key userKey = KeyFactory.createKey(PersonBuilder.ENTITY_NAME, userGithub);
-    System.out.println(userKey);
 
+    String userID =
+        request.getParameter("userID") != null
+            ? request.getParameter("userID")
+            : AuthServlet.getAuthorizedUser().getUserId();
+
+    Key userKey;
+    try {
+      userKey = KeyFactory.createKey(PersonBuilder.ENTITY_NAME, userID);
+    } catch (Exception e) {
+      ErrorResponse.sendJsonError(
+          response,
+          e.getMessage() + "/n/n" + e.getStackTrace(),
+          HttpServletResponse.SC_BAD_REQUEST,
+          "An error occured while getting the User's profile");
+      return;
+    }
```

Comment:
Is the HTTP 400 error code the most appropriate error code here? Looking at the user message here, I think a 500 error (or some server error) is more appropriate. A 400 error implies that the error can be fixed by actions from the user.

Reply:
Also I'm not sure if I understand why there is a try-catch here. It doesn't seem like [createKey](https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/KeyFactory.html#createKey-java.lang.String-java.lang.String-) throws any meaningful errors.

Reply:
It throws an error If the UserID is null. The Exception is not in the documentation for some reason.